handler.go

```diff
@@ -449,35 +450,30 @@ func populateTmpDir(r *http.Request, rmDir bool, dirpath string) (string, *gz.Er
 	}
 
 	// First check if all files are in the same root dir and if we should rm it.
-	var outDir string
-	if !strings.Contains(files[0].Filename, "/") {
-		// No folder in first file, then there is no common folder
-		rmDir = false
-	} else {
-		// Find out the folder name
-		if strings.HasPrefix(files[0].Filename, "/") {
-			outDir = "/" + strings.SplitAfter(filepath.Clean(files[0].Filename), "/")[1]
-		} else {
-			outDir = strings.SplitAfter(filepath.Clean(files[0].Filename), "/")[0]
-		}
-		for i := 0; i < fLen && rmDir; i++ {
-			rmDir = strings.HasPrefix(files[i].Filename, outDir)
-		}
+	rmDir, outDir, em := getOuterDir(files, rmDir)
+	if em != nil {
+		return "", em
 	}
 
 	// Process files
 	for _, fh := range files {
+		fn, err := extractFilepath(fh)
+		if err != nil {
+			continue
+		}
+		if len(fn) == 0 {
+			continue
+		}
 		file, err := fh.Open()
+		if err != nil {
+			return "", gz.NewErrorMessageWithBase(gz.ErrorForm, err)
+		}
```

Comment:
Why is this happening before the deferral? Is `file = nil` and panics if we call `Close()`?